aom/src/main/java/com/nedap/archie/aom/utils/AOMUtils.java

```diff
@@ -338,6 +341,42 @@ public static int getMaximumIdCode(int specializationDepth, String prefix, Colle
         return maximumIdCode;
     }
 
+    public static boolean isPathInArchetypeOrRm(MetaModel metaModel, String path, Archetype template) {
+        AOMPathQuery aomPathQuery = new AOMPathQuery(path);
+        PartialMatch partial = aomPathQuery.findPartial(template.getDefinition());
+        if(partial.getRemainingPath().isEmpty() || partial.getRemainingPath().equals("/")) {
+            return true;
+        } else {
+            if(isArchetypePath(partial.getRemainingPath())) {
+                // the remaining path is an archetype path, so cannot be found purely in the RM without
+                //further constraints
+                return false;
+            }
+            //we have a partial match left, search for it in the RM
+            if(partial.getFoundObjects().isEmpty()) {
```

Comment:
hmm, this can't happen since it always returns at least the root node, so this case should probably be removed.